tests/test_zg.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import numpy as np
+import matplotlib.pyplot as plt
+import networkx as nx
+from numpy import array_equal, allclose
+from numpy.linalg import norm
+
+from graphstats.embed.svd import selectDim
+from scipy.linalg import orth
+
+def generate_data(n=10, elbows=3, seed=1):
+    """
+    Generate data matrix with a specific number of elbows on scree plot
+    """
+    np.random.seed(seed)
+    x = np.random.binomial(1,.6,(n**2)).reshape(n,n)
+    xorth = orth(x)
+    d = np.zeros(xorth.shape[0])
+    for i in range(0,len(d), int(len(d)/(elbows+1))):
+        d[:i] += 10
+    A = xorth.T @ np.diag(d) @ xorth
+    return A,d
+
+def test_zg_1_elbow():
+    # Should get all ones
+    data,l = generate_data(10,3)
+    elbows, likelihoods, sing_vals, all_likelihoods = selectDim(data, 3)
```

Comment:
Could we add more tests, typechecking at the very least? 

Also, it would be good to demonstrate that when using ASE or any of the other functions that call svd, those functions can now use dimselect. Idk if those tests should go here or in those respective test files but I think we should have them. As is, I don't think they can actually utilize dimselect

Reply:
Yea agree. This really needs more tests.